eZ/Publish/Core/Repository/Helper/DomainMapper.php

```diff
@@ -136,15 +136,17 @@ public function buildDomainFields(array $spiFields, $contentType)
 
         $fields = array();
         foreach ($spiFields as $spiField) {
-            $fields[] = new Field(
-                array(
-                    'id' => $spiField->id,
-                    'value' => $this->fieldTypeRegistry->getFieldType($spiField->type)
-                        ->fromPersistenceValue($spiField->value),
-                    'languageCode' => $spiField->languageCode,
-                    'fieldDefIdentifier' => $fieldIdentifierMap[$spiField->fieldDefinitionId],
-                )
-            );
+            if (array_key_exists($spiField->fieldDefinitionId, $fieldIdentifierMap)) {
```

Comment:
@Nattfarinn could you checke the patch like this *(return/continue early pattern)*:

```php
foreach ($spiFields as $spiField) {
    // We ignore fields in content not part of the content type
    if (!isset($fieldIdentifierMap[$spiField->fieldDefinitionId])) {
        continue;
    }
```